Add 아마존 캐러샐 html, css, 애니메이션 기능 구현

hyewon3938 opened the pull request:
깃페이지 : https://hyewon3938.github.io/amazonCarousel/amazon.html

클래스로 나누는 부분은 진행 중인데 생각보다 시간이 지체되어서 
우선 무한으로 슬라이드 되는 캐러셀 애니메이션까지 구현한 코드 먼저 PR합니다

~~그리고 라이브서버로 확인하면서 만들 때는 분명히 이상이 없었는데 
같은 코드를 깃 페이지로 만들고나니 css가 약간 뒤틀렸네요
이 부분도 추가로 수정하겠습니다 
정말 css는 제 마음대로 안 되는 것 같아요~~

**=> reset.css 파일 업로드를 안 해서 생긴 문제였습니다. 파일 업로드 후 이제 잘 실행됩니다!**

코드를 짜고 구현하는 시간이 많이 느린 편인 것 같습니다
속도를 조금 더 높일 수 있도록 노력해보겠습니다

[추가해야 할 부분]
- 클래스로 나누기
- animation 구현 부분 main.js 에서 분리해주기

Commit:
Add 아마존 캐러샐 html, css, 애니메이션 기능 구현

Commit:
feat 아마존 캐러샐 html, css, 애니메이션 기능 구현

crongro commented on amazon.css:
```diff
@@ -0,0 +1,94 @@
+    transform: translate(-450px,-50px);
+    overflow: hidden;
+}
+.slide_item_wrap{
```
css classname을 잘 짓고 계시네요.

crongro commented on amazon.css:
```diff
@@ -0,0 +1,94 @@
+    left: 60px;
+    z-index: 1;
+}
+.next_btn{
```
아래처럼 공통부분을 하나의 클래스로 만들어서 같이 쓰는 것도 좋죠. 

```css
.btn {
    position : absolute;
    top: 70%;
    z-index: 1;
}
.prev_btn{
  left:60px;
}
```

crongro commented on amazon.html:
```diff
@@ -0,0 +1,85 @@
+                    </div>
+                </li>
+
+                <li class="slide_item">
```
이번주 미션은 li로 되어 있는 데이터를 이렇게 HTML페이지에 미리 추가하지 않고, 서버에서 받아서 동적으로 추가해야할거에요.

crongro commented on amazonAnimation.js:
```diff
@@ -0,0 +1,114 @@
+const [prev, next] = document.querySelectorAll("button");
```
일단 시작은 이렇게 전역변수로 만들어도 됩니다.
그 다음으로 이제 class로 리팩토링 하는 단계를 밟아나가시면 될 듯.

묶음기능들을 클래스의 메서드로 만들어서 넣고, 
같이 쓰는 속성등  this.listLenth = ... 이런식으로 constructor에서 추가하고요.

crongro reviewed (changes requested):
처음에는 이렇게 구현해도 됩니다!
현재 기능을 class로 리팩토링 해보는 단계로 진행해주시면 될 거 같아요. 
도전해보세요! 

기능개발 후에 push 하시면 PR에 업데이트 될거에요.

Commit:
Merge branch 'animation' of https://github.com/hyewon3938/javascript-amazon into animation

Commit:
feat 파일에 저장된 데이터를 캐러샐 아이템 요소로서 동적으로 html에 추가되도록 함

mockData.js에 객체형태로 저장된 데이터를 navigation과 button, slide로 나누어서 각각 html태그로 추가되도록 구현했다. 슬라이드 매니메이션을 적용하는 자바스크립트 코드는 아직 그냥 main.js 아래에 두었는데 추가적으로 분리해줘야할 것 같다. 애니메이션 적용 코드를 클래스로 나눈 다음에 분리해줘야겠다.

crongro commented on carousel.js:
```diff
@@ -0,0 +1,68 @@
+
+}
+
+class Nav {
```
Nav보다 명확한 이름으로 지어보세요.

crongro commented on carousel.js:
```diff
@@ -0,0 +1,68 @@
+// const mockData = require("./mockData");
+
+class Carousel {
```
Carousel은 재사용되는 형태는 아니고, 
여러가지 객체를 생성해서 서비스가 동작되는 시작포인트 같네요.
그렇다면 이런 역할은 Class형태는 아니여도 되긴합니다.

crongro commented on carousel.js:
```diff
@@ -0,0 +1,68 @@
+    this.slideData = slideData;
+  }
+
+  render() {
```
render라는 이름은 좋은거 같아요.
makeHTML 이라는 이름도 해도 되고요.

crongro reviewed (approved):
클래스 형태로 view영역이 바뀐거 같네요.
순수한 view형태를 만든것은 좋긴한데요,  event 등록부분 코드도 Class안으로 옮길 수 있겠죠. 

Class에 기능을 더  가득하게 만드셔도 될 거 같아요.

참고하시고 계속 리팩토링!과 기능개발해보세요. 
여기까지 승인할테니 이어서 다시 pr보내시면 됩니다.
